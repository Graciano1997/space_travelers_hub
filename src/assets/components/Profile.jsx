import { useSelector } from 'react-redux';
import style from '../style/Profile.module.css';

function Profile() {
  const { rocketsArray } = useSelector((state) => state.rockets);
  const reservedRockets = rocketsArray.filter((rocket) => rocket.reserved);

  return (
    <div className={style.profileContainer}>
      <div className={style.profileMissions}>
        <h1>My Missions</h1>
        <div className={style.missionContainer}>
          <p>
            Lorem, ipsum dolor sit a
            met consectetur adipisicing elit. Dolore distinctio doloribus, num?
          </p>
          <p>
            Lorem, ipsum dolor sit a
            met consectetur adipisicing elit. Dolore distinctio doloribus, num?
          </p>
          <p>
            Lorem, ipsum dolor sit a
            met consectetur adipisicing elit. Dolore distinctio doloribus, num?
          </p>
          <p>
            Lorem, ipsum dolor sit a
            met consectetur adipisicing elit. Dolore distinctio doloribus, num?
          </p>
        </div>
      </div>
      <div className={style.profileRockets}>
        <h1>My Rockets</h1>
        <div className={style.rocketsContainer}>
          {(reservedRockets.length === 0) && (
            <div>
              <h3>You don&apos;t have any reserved Rockets! ðŸš€ðŸ˜‰ðŸ˜Š</h3>
            </div>
          )}

          {reservedRockets.map((rocket, index) => (
            <div key={{ index }}>
              <h3>
                {' '}
                {`${rocket.rocket_name}ðŸš€`}
              </h3>
            </div>
          ))}

        </div>
      </div>
    </div>
  );
}

export default Profile;
