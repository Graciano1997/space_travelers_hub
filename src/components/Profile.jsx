import Alert from 'react-bootstrap/Alert';
import { useSelector } from 'react-redux';
import ListGroup from 'react-bootstrap/ListGroup';
import style from '../assets/style/Profile.module.css';

function Profile() {
  const { rocketsArray } = useSelector((state) => state.rockets);

  const { missions } = useSelector((state) => state.missions);

  return (
    <div className={style.profileContainer}>
      <div className={style.profileMissions}>
        <h1>My Missions</h1>
        <ListGroup className={style.missionContainer}>
          {(missions.filter((mission) => mission.reserved).length === 0) && (
            <ListGroup.Item><Alert variant="info">You don&apos;t have any reserved Misssions 👨‍🚀👩🏿‍🚀 </Alert></ListGroup.Item>
          )}

          {
            missions.filter((mission) => mission.reserved)
              .map((mission) => (
                <ListGroup.Item key={mission.mission_id}>
                  {' '}
                  {`${mission.mission_name}👩🏿‍🚀`}
                </ListGroup.Item>
              ))
          }
        </ListGroup>
      </div>

      <div className={style.profileRockets}>
        <h1>My Rockets</h1>
        <ListGroup className={style.rocketsContainer}>

          {(rocketsArray.filter((rocket) => rocket.reserved).length === 0) && (
            <ListGroup.Item><Alert variant="info">You don&apos;t have any reserved Rockets! 🚀😉😊</Alert></ListGroup.Item>
          )}

          {rocketsArray.filter((rocket) => rocket.reserved)
            .map((rocket) => (
              <ListGroup.Item key={rocket.id} title="rocketItem">
                {' '}
                {`${rocket.rocket_name}🚀`}
              </ListGroup.Item>
            ))}

        </ListGroup>
      </div>
    </div>
  );
}

export default Profile;
