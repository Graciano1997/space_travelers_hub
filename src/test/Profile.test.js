import { Provider } from 'react-redux';
import { render, screen } from '@testing-library/react';
import store from '../redux/store';
import '@testing-library/jest-dom/extend-expect';
import Profile from '../components/Profile';

function MockingRocketsProfile() {
  return (
    <Provider store={store}>
      <Profile />
    </Provider>
  );
}

describe('testing the Profile', () => {
  test('see how it renders ', () => {
    const { tree } = render(<MockingRocketsProfile />);
    expect(tree).toMatchSnapshot();
  });

  test('mission', () => {
    render(<MockingRocketsProfile />);
    const word = screen.getByText(/mission/i);
    expect(word).toBeInTheDocument();
  });
});
