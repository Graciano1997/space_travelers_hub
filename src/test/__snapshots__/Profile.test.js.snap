// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the Profile When we Reserved Rockets 1`] = `undefined`;

exports[`testing the Profile mission 1`] = `undefined`;

exports[`testing the Profile see how it renders  1`] = `undefined`;
